packages/storybook/src/stories/template-generator.ts

```diff
@@ -0,0 +1,107 @@
+import { Story } from '@storybook/web-components';
+import docsJson from '../../elements-stencil-docs';
+
+/**
+ * A utility class that helps with creating type-safe stories
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export class TemplateGenerator<Component extends Object> {
+  private readonly tagName: string;
+  private readonly templateFn: Story<Component>;
+
+  constructor(tagName: string, templateFn: Story<Component>) {
+    this.tagName = tagName;
+    this.templateFn = templateFn;
+  }
+
+  /**
+   * Generates the default template story.
+   */
+  public generatePlaygroundStory() {
+    return this.generateTemplate();
+  }
+
+  /**
+   * Generates a story based of a key-value-pair and automatically adds the documentation of the property.
+   *
+   * @param key The key of the property
+   * @param value The value of the property which is assignable to `key`
+   */
+  public generateStory<Key extends keyof Component>(
+    key: Key,
+    value: Component[Key]
+  ) {
+    const argument: Component = ({ [key]: value } as unknown) as Component;
+
+    return this.generateTemplate(argument, key);
+  }
+
+  /**
+   * Generates a new story based of multiple arguments.
+   * Adds a description of the parameter which is provided by `prop`.
+   *
+   * @param args Properties to assign to the template
+   * @param prop The name of the "main" property the stories is written for
+   */
+  public generateStoryWithArgs(args: Component, prop: keyof Component) {
+    return this.generateTemplate(args, prop);
+  }
+
+  /**
+   * A utility function to inherit the args of one or more stories.
+   *
+   * @param mainStory The main story which arguments have the highest priority
+   * @param storiesToReuse One or more stories which args should be inherited. The last one has the highest priority (still below `mainStory`).
+   */
+  public extendStory(
```

Comment:
Not quite sure about that one. Another option would be to introduce our own class `Story` which extends the storybook type. In there we could provide a method like `inherit`. Something like this:
```ts
class OurStory extends Story<Component> {
 // ...

  inherit(story: Story<Component>) {
        this.args = Object.assign({}, this.args, story.args);
        return this;
  }
}
```

That would allow us to write our stories like this:
`export const Selection = template.generateStory('selection', true).inherit(Checked).inherit(...)`

What do you think @pfecht ?

Reply:
This whole inheritance concept doesn't look that intuitive to me. Is the goal to inherit only the default state of args (because the template is defined be the template builder, isn't?) 

Then I wouldn't support this and provide an additional `defaultValues` field or so to the `generateStory`:

`generateStory('indeterminate', true, {'checked': true})`

What do you think?

Reply:
The goal is to minimize copy-pasting of other stories.
Imagine you want to have a Story A and Story B.

Story A is quite complex and has five properties that need to be set. Story B has the same properties but with one exception. You would have to copy the exact same properties to Story B and in the long run that would result in higher maintenance costs.